app/scripts/patient-dashboard/services/notes.service.js

```diff
@@ -0,0 +1,366 @@
+      var startIndex = startIndex || 0;
+      
+      // Limit to 10 notes if not specified (i.e 10 encounters going backward)
+      var limit = limit || DEFAULT_NO_NOTES;
+      
+      // Get HIV Summary & vitals
+      var hivPromise = EtlRS.getHivSummary(patientUuid, startIndex, limit);
+      var vitalPromise = EtlRS.getVitals(patientUuid, startIndex, limit);
+      
+      var encParams = {
+        patientUuid: patientUuid,
+        rep: encounterRepresentation
+      };
+      var encPromise = EncounterRS.getPatientEncounters(encParams);
+        
+      $q.allSettled([hivPromise, vitalPromise, encPromise]).then(function(data) {
+        var hivSummaries = null;
+        var vitals = null;
+        var encounters = null;
+        
+        if(data[0].status === 'fulfilled') {
+          var hivSummaries = data[0].value.result;
+        } else {
+          throw new Error('error fetching summaries');
+        }
+        
+        if(data[1].status === 'fulfilled') {
+          var vitals = data[1].value.result;
+        } else {
+          throw new Error('error fetching vitals');
+        }
+        
+        if(data[2].status === 'fulfilled') {
+          var encounters = data[2].value;
+        } else {
+          throw new Error('error fetching encounters');
+        }
+        
+        if(!hivSummaries || _.isEmpty(hivSummaries)) {
+          $log.error('Could not generate notes because no hiv summaries have' +
+                      ' been returned');
+          throw new Error('No summaries returned');
+        } else {
+          // TODO: Need to determine whether it is wise to fail if others are 
+          // not available as well.
+          
+          // Create array of models
+          var hivModels = HivSummaryModel.toArrayOfModels(hivSummaries);
+          
+          var summaryDateGrouped = {};
+          _.each(hivModels, function(model) {
+            var key = moment(model.encounterDatetime()).format('MMM_DD_YYYY');
+            model.ordering = encOrder[model.encounterType()] || encOrder.other;
+            if(Array.isArray(summaryDateGrouped[key])) {
+              summaryDateGrouped[key].push(model);
+            } else {
+              summaryDateGrouped[key] = [model];
+            }
+          });
+          
+          // Sort the groups according to ordering column
+          _.each(summaryDateGrouped, function(group) {
+            group.sort(function compare(x, y) {
+              return x.ordering - y.ordering;
+            });
+          });
+          
+          // Now pick the most preferred encounter in every group.
+          var massagedHivModels = _.map(summaryDateGrouped, function(group) {
+            return _.first(group);
+          });
+          
+          // Deal with vitals.
+          if(vitals && !_.isEmpty(vitals)) {
+            var vitalModels = VitalModel.toArrayOfModels(vitals);
+            
+            // Create a date grouped representation
```

Comment:
In general, we will need to change this flow to encounter based. For now don't worry about multiple vitals per visit

Sent from my iPhone

> On 22 Apr 2016, at 10:17 AM, nkimaina <notifications@github.com> wrote:
> 
> In app/scripts/patient-dashboard/services/notes.service.js:
> 
> > +          _.each(summaryDateGrouped, function(group) {
> > +            group.sort(function compare(x, y) {
> > +              return x.ordering - y.ordering;
> > +            });
> > +          });
> > +          
> > +          // Now pick the most preferred encounter in every group.
> > +          var massagedHivModels = _.map(summaryDateGrouped, function(group) {
> > +            return _.first(group);
> > +          });
> > +          
> > +          // Deal with vitals.
> > +          if(vitals && !_.isEmpty(vitals)) {
> > +            var vitalModels = VitalModel.toArrayOfModels(vitals);
> > +            
> > +            // Create a date grouped representation
> What if there are more than one vitals per day?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
>